components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/factory/KeyManagerHolder.java

```diff
@@ -71,51 +71,54 @@ public static void addKeyManagerConfiguration(String tenantDomain, String name,
             log.error("Key Manager " + name + " already initialized in tenant " + tenantDomain);
         }
         if (keyManagerConfiguration.isEnabled()) {
-            KeyManager keyManager = null;
-            JWTValidator jwtValidator = null;
-            APIManagerConfiguration apiManagerConfiguration =
-                    ServiceReferenceHolder.getInstance().getAPIManagerConfigurationService()
-                            .getAPIManagerConfiguration();
-            String defaultKeyManagerType =
-                    apiManagerConfiguration.getFirstProperty(APIConstants.DEFAULT_KEY_MANAGER_TYPE);
-            KeyManagerConnectorConfiguration keyManagerConnectorConfiguration =
-                    ServiceReferenceHolder.getInstance().getKeyManagerConnectorConfiguration(type);
-            if (keyManagerConnectorConfiguration != null) {
-                if (StringUtils.isNotEmpty(keyManagerConnectorConfiguration.getImplementation())) {
-                    try {
-                        keyManager = (KeyManager) Class
-                                .forName(keyManagerConnectorConfiguration.getImplementation()).newInstance();
-                        keyManager.setTenantDomain(tenantDomain);
-                        if (StringUtils.isNotEmpty(defaultKeyManagerType) && defaultKeyManagerType.equals(type)){
-                            keyManagerConfiguration.addParameter(APIConstants.KEY_MANAGER_USERNAME,
-                                    apiManagerConfiguration.getFirstProperty(APIConstants.API_KEY_VALIDATOR_USERNAME));
-                            keyManagerConfiguration.addParameter(APIConstants.KEY_MANAGER_PASSWORD,
-                                    apiManagerConfiguration.getFirstProperty(APIConstants.API_KEY_VALIDATOR_PASSWORD));
+            if (!StringUtils
+                    .equals(keyManagerConfiguration.getTokenType(), APIConstants.KeyManager.TOKEN_TYPE_EXCHANGED)
+                    || StringUtils.equals(keyManagerConfiguration.getTokenType(), StringUtils.EMPTY)) {
```

Comment:
I think it's better to set a default value for `tokenType` in `keyManagerConfiguration` similar to what we did in OpenAPI models. Also, we can add java enums for `tokenType` if possible. WDYT?

Reply:
+1. Fixed via 7411bd266c60ffe5dc3f174da2818c481bfca20f